### PR TITLE
Scala/Room integration. Migrate SQLiteDatabase to SupportSQLiteDatabase

### PR DESCRIPTION
**Pull request by BradleyWire**
### Issues

When we decide to turn on the `KOTLIN_MIGRATION` flag and trigger the migration once we enter the settings screen. Once you put the application into the background - when you load the app up again you'll be met with a blank screen once we've entered the application

### Causes

This is due to the database migrations and initialisation relying on manual processes with a constant db version to restore data on login.

### Solutions

Make the old migrations compatible with the new Room migrations so we can bridge the gap somehow by migrating over SQLiteDatabase and the old SQLOpenHelper (DaoDB) to a new SupportSQLiteDatabase and SupportSQLOpenHelper which Room builds a foundation on to ensure there is some kind of compatibility between the two. 

## Notes

This isn't the complete issue resolved, with @gizemb's changes on top of this we'll be able to enable a dynamic migration of versions that are both compatible with the new room migrations and the old database migrations (which we still need the logic surrounding that in some use-cases) 

#### APK
[Download build #1290](http://10.10.124.11:8080/job/Pull%20Request%20Builder/1290/artifact/build/artifact/wire-dev-PR2622-1290.apk)
[Download build #1291](http://10.10.124.11:8080/job/Pull%20Request%20Builder/1291/artifact/build/artifact/wire-dev-PR2622-1291.apk)
[Download build #1314](http://10.10.124.11:8080/job/Pull%20Request%20Builder/1314/artifact/build/artifact/wire-dev-PR2622-1314.apk)

### CODE REVIEW

File: wire-android-sync-engine/zmessaging/src/main/scala/com/waz/db/ZMessagingDB.scala [hunk 1]
```diff
@@ -67,7 +66,7 @@ class ZMessagingDB(context: Context, dbName: String, tracking: TrackingService)
 }
 
 object ZMessagingDB {
-  val DbVersion = 126
+  val DbVersion = UserDatabase.VERSION
```
Comment (BradleyWire) [inline wire-android-sync-engine/zmessaging/src/main/scala/com/waz/db/ZMessagingDB.scala]: This is temporary until we integrate @gizemb's fix into this.

**Comment by BradleyWire** [comment]
Will update test cases.

File: app/src/androidTest/scala/com/waz/background/WorkManagerSyncRequestServiceSpec.scala [hunk 2]
```diff
@@ -1,178 +0,0 @@
-/**
```
Comment (mejdoo) [inline app/src/androidTest/scala/com/waz/background/WorkManagerSyncRequestServiceSpec.scala]: 👍  great ! But it would be better if done in a separate PR

File: wire-android-sync-engine/zmessaging/src/main/scala/com/waz/db/package.scala [hunk 3]
```diff
@@ -126,22 +126,19 @@ package db {
...
-        db.acquireReference()
-        try {
-          db.getThreadSession.beginTransaction()
-        }
-        finally db.releaseReference()
+      override def beginReadTransaction(db: DB): Unit = try reflectiveBegin(db) catch {
+        case _: Exception => db.beginTransactionNonExclusive()
       }
+
+      private def reflectiveBegin(db: DB): Unit = db.beginTransaction()
```
Comment (makingthematrix) [inline wire-android-sync-engine/zmessaging/src/main/scala/com/waz/db/package.scala]: You can inline this method, and then, in `beginReadTransaction` use `Try` instead of `try/catch`:
`Try(db.beginTransaction()).recover { case _ => db.beginTransactionNonExclusive() }`

**Comment by mejdoo** [comment]
Good job. Hope this will fix the issue

File: app/src/androidTest/scala/com/waz/background/WorkManagerSyncRequestServiceSpec.scala (hunk 2 above)
Comment (BradleyWire) [reply app/src/androidTest/scala/com/waz/background/WorkManagerSyncRequestServiceSpec.scala]: I was debating that myself, but, there's no urgency for the tests to go into develop and it doesn't pollute this PR's scope too much as we are changing the tests.

**Comment by BradleyWire** [comment]
> Good job. Hope this will fix the issue

It doesn't unfortunately but we are working on a fix as we've found the issue.

File: wire-android-sync-engine/zmessaging/src/main/scala/com/waz/model/package.scala [hunk 4]
```diff
@@ -17,18 +17,14 @@
...
 import com.google.protobuf.nano.{CodedInputByteBufferNano, MessageNano}
 import com.waz.model.nano.Messages
 import com.waz.utils.crypto.AESUtils
 import com.waz.utils.{JsonDecoder, JsonEncoder, returning}
-import io.circe.{Decoder, Encoder}
 import org.json.JSONObject
 
-import scala.language.implicitConversions
 import scala.concurrent.duration.FiniteDuration
+import scala.language.implicitConversions
```
Comment (makingthematrix) [inline wire-android-sync-engine/zmessaging/src/main/scala/com/waz/model/package.scala]: I think we don't need implicitConversions anywhere here.

File: wire-android-sync-engine/zmessaging/src/main/scala/com/waz/db/package.scala [hunk 5]
```diff
@@ -126,22 +126,19 @@ package db {
...
 
-      override def beginReadTransaction(db: DB): Unit = try reflectiveBegin(db) catch { case _: Exception => db.beginTransactionNonExclusive() }
-
-      private def reflectiveBegin(db: DB): Unit = {
-        db.acquireReference()
-        try {
-          db.getThreadSession.beginTransaction()
-        }
-        finally db.releaseReference()
+      override def beginReadTransaction(db: DB): Unit = try reflectiveBegin(db) catch {
```
Comment (gizemb) [inline wire-android-sync-engine/zmessaging/src/main/scala/com/waz/db/package.scala]: maybe we can get rid of reflectiveBegin method here. not that important though.
